lib/breadbox/s3_client.rb

```diff
@@ -13,10 +13,11 @@ def s3_bucket_object
 
     def upload(options = {})
       path         = options[:path]
+      filename     = options[:filename]
       file         = options[:file]
       acl          = options[:public] ? :public_read : nil
       content_type = options[:content_type]
-      filepath = filepath_from_paths_and_file(root_path, path, file)[1..-1]
+      filepath = filepath_from_paths_and_filename(root_path, path, filename || File.basename(file))[1..-1]
```

Comment:
My initial feedback on this, is that the original `filepath_from_paths_and_file(root_path, path, file)[1..-1]` was already hairy due to the removal of the first array element inline while assigning to filepath... but when you put in `filename || File.basename(file)` it starts to look even more gnarly. It's fine for now, just food for thought.